primefaces/src/main/resources/META-INF/resources/primefaces/forms/forms.selectcheckboxmenu.js

```diff
@@ -121,9 +122,12 @@ PrimeFaces.widget.SelectCheckboxMenu = PrimeFaces.widget.BaseWidget.extend({
         this.menuIcon = this.jq.children('.ui-selectcheckboxmenu-trigger');
         this.triggers = this.jq.find('.ui-selectcheckboxmenu-trigger, .ui-selectcheckboxmenu-label');
         this.disabled = this.jq.hasClass('ui-state-disabled');
-        this.inputs = this.jq.find(':checkbox');
-        this.panelId = this.id + '_panel';
+        //Only include checkboxes with an ID that starts like ID of the menu!
+        this.inputs = this.jq.find(':checkbox').filter('[id^="' + this.id.replace(/:/g,"\\:") + '"]');
```

Comment:
OK. Before it was `this.inputs = this.jq.find(':checkbox');`. But it's not exact enough because there is a toggler checkbox. Maybe I also try by class.

Reply:
It should be safe to search in the "hidden" `div` that contains the actual checkbox input elements:

`this.inputs = this.jq.filter('.ui-helper-hidden').find(':checkbox');`

What do you think?

Reply:
Better!

Reply:
@jepsar FYI:
My proposed fix did not work and resulted in failed integration tests. Apparently because for the target element the CSS `display` is set to `none` initially which makes it invisible to the filter.
So I used this instead:
`this.jq.children('div.ui-helper-hidden:not(.ui-input-overlay)').children(':checkbox');`